src/Util.cpp

```diff
@@ -411,62 +457,145 @@ std::vector<double> ale::getDistortionCoeffs(json isd) {
   return coefficients;
 }
 
-std::vector<double> ale::getSunPositions(json isd) {
-  std::vector<double> positions;
+std::vector<Vec3d> getJsonVec3dArray(json obj) {
+  std::vector<Vec3d> positions;
   try {
-    json jayson = isd.at("sun_position");
-    for (auto &location : jayson.at("positions")) {
-      positions.push_back(location[0].get<double>());
-      positions.push_back(location[1].get<double>());
-      positions.push_back(location[2].get<double>());
+    for (auto &location : obj) {
+      Vec3d vec(location[0].get<double>(),location[1].get<double>(), location[2].get<double>() );
+      positions.push_back(vec);
     }
   } catch (...) {
-    throw std::runtime_error("Could not parse the sun positions.");
+    throw std::runtime_error("Could not parse the 3D vector array.");
   }
   return positions;
 }
 
-std::vector<double> ale::getSensorPositions(json isd) {
-  std::vector<double> positions;
+
+std::vector<Rotation> getJsonQuatArray(json obj) {
+  std::vector<Rotation> quats;
   try {
-    json jayson = isd.at("sensor_position");
-    for (auto &location : jayson.at("positions")) {
-      positions.push_back(location[0].get<double>());
-      positions.push_back(location[1].get<double>());
-      positions.push_back(location[2].get<double>());
+    for (auto &location : obj) {
+      Rotation vec(location[0].get<double>(),location[1].get<double>(), location[2].get<double>(), location[2].get<double>() );
```

Comment:
Should last index be 3, rather than 2?

Reply:
If so, I also suggest updating the test so that it could catch this kind of thing.

Reply:
definitely a bug, fixed and updated tests.